src/lib/hooks/useCurrency.ts

```diff
@@ -105,8 +102,7 @@ export function useCurrencyFromMap(tokens: TokenMap, currencyId?: string | null)
 
   const token = useTokenFromMapOrNetwork(tokens, isNative ? undefined : shorthandMatchAddress ?? currencyId)
 
-  const chainAllowed = chainId && isChainAllowed(connector, chainId)
-  if (currencyId === null || currencyId === undefined || !chainAllowed) return null
+  if (currencyId === null || currencyId === undefined) return null
```

Comment:
this also seems unrelated? would be nice to keep the diff clean

Reply:
i could have the function just return `true` and leave all this if we wanted to

Reply:
reverted

Reply:
this actually was a bug, i needed to check ALLOWED_CHAIN_IDS.

Reply:
i'd like to rename this to `isChainSupported` separately also